pkg/authenticator/k8s/authenticator.go

```diff
@@ -48,9 +48,27 @@ const (
 	nameTypeIP    = 7
 )
 
+// Init the authenticator struct
+func (auth *Authenticator) Init(config *common.ConfigurationInterface) (common.AuthenticatorInterface, error) {
+	var cfg = (*config).(*Config)
+	authn, err := New(*cfg)
+	if err != nil {
+		return nil, fmt.Errorf(log.CAKC019)
+	}
+
+	return authn, nil
+}
+
+func (auth *Authenticator) InitWithAccessToken(config *common.ConfigurationInterface, token access_token.AccessToken) (common.AuthenticatorInterface, error) {
```

Comment:
I'm not sure about the name of the method. I think we need something other than `Init` that better captures what this method is doing.

This should also probably be a free standing function (not a method) since it doesn't use the auth struct at all.

Reply:
Lets discuss in the meeting